doc/dev/developer_guide/tests-unit-tests.rst

```diff
@@ -48,6 +48,38 @@ functions, which are themselves defined in
 others are binaries that are compiled during the build process.  The
 ``add_ceph_test`` function is used to declare unit test scripts, while
 ``add_ceph_unittest`` is used for unit test binaries.
+``add_tox_test`` declare tests that require ``tox`` for testing and sets up 
+the full environment for this during testing.
+::
+
+        add_ceph_test(test_name test_path <args> <options>)
+        add_ceph_unittest(test_name <args> <options>)
+        add_tox_test(test_name <args> <options>)
+        args: any parameter, not being an options keyword
+        options:
+            [COST <float>]
+               Order of testing is set by COST. 
+               Highest COST are tested first.
+            [TIMEOUT sec]
+               Restrict the max runtime of this test to <sec>
+            [LABELS "labels"]
+               Append "labels" to a test.
+               Labels can be selected with ctest -L or -LE
+
+Since COST is used to determine the testing order, values can be chosen rather
+arbitrarily. Currently long running programs are given a value equal to the
+execution time in a OpenStack VM. The idea is to first run the long standing 
+tests, and then fill up free cores with programs with shorter execution time.
+Turning this into a bin-packing challenge, So the exact value is not important,
+it is about the relative COST value.
```

Comment:
`an OpenStack`.  But conceptually I'm a bit unclear, how is OpenStack involved?  Wouldn't runtime in an OpenStack instance be a function of the aggregate's overcommit policy, the instance flavor, and and the underlying CPU model?  Where does the numerical COST come from?  Historical data? Enquiring minds want to know.

"The idea is to first run the long-running tests, and then fill unused cores with tests with shorter execution time.
This is effectively a bin-packing challenge, so the _exact_ values are not important, bur rather _relative_ COST value.

Reply:
> `an OpenStack`. But conceptually I'm a bit unclear, how is OpenStack involved? Wouldn't runtime in an OpenStack instance be a function of the aggregate's overcommit policy, the instance flavor, and and the underlying CPU model? Where does the numerical COST come from? Historical data? Enquiring minds want to know.
> 
> "The idea is to first run the long-running tests, and then fill unused cores with tests with shorter execution time. This is effectively a bin-packing challenge, so the _exact_ values are not important, bur rather _relative_ COST value.

The main purpose here is to prevent that run in a multi-core environment and try to schedule to very long running jobs at the end of test-run, since that could extend the test-run time with like 5 - 10 minutes.

'Openstack ' refers to the fact that I was running these builds in an OpenStack system. But like I indicated `COST` is only used to steer the bin-packing algorithm, no use in getting exact numbers, order of magnitude will already work.
And I used the times that I found in my runs.